atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionManager.java

```diff
@@ -102,4 +102,9 @@
      * @return value returned by task
      */
     <T, E extends Exception> T runTaskReadOnly(TransactionTask<T, E> task) throws E;
+
+    /**
+     * Performs non-destructive cleanup when the transaction manager is no longer needed.
+     */
+    void close();
```

Comment:
should `@Override` `AutoCloseable#close()`. we should also consider the implications of what calling other txn manager methods after a transaction manager has been closed -- they should probably throw `IllegalStateException` indicating that operations cannot be performed once a transaction manager has been closed, and similar `Preconditions.checkState(!isClosed, "Cannot perform operation on closed KeyValueService")` sanity checks for KVS should probably be added.